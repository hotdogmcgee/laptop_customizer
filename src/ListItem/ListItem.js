import React from 'react'
import ListItemOption from '../ListItem_option/ListItemOption.js'

class ListItem extends React.Component {
    constructor(props) {
        super(props)
    }

    
    //render a component that displays two options
    render() {

        const { featuresItem } = this.props
        const ListItem_item = Object.keys(featuresItem).map((item, index) => 
                <ListItemOption
                    item={item}
                    updateFeature={this.props.updateFeature}
                    key={index}
                 />

            );


            return (
                <div className="feature" key={featuresItem.name}>
                <div className="feature__name">{featuresItem.name}</div>
                <ul className="feature__list">
                    { ListItem_item }
                </ul>
                </div>
            )
    }
}


export default ListItem