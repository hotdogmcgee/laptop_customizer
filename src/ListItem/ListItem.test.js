import React from 'react';
import ReactDOM from 'react-dom';
// import renderer from 'react-test-renderer';
import ListItem from './ListItem.js'

describe('ListItem component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ListItem />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

//   it('renders the UI as expected', () => {
//     const tree = renderer
//       .create(<ListItem item="item" index="index" updateFeature=""/>)
//       .toJSON();
//     expect(tree).toMatchSnapshot();  
//   });

});