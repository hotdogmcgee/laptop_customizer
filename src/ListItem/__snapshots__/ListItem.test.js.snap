// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders the UI as expected 1`] = `
<div
  className="feature"
>
  <div
    className="feature__name"
  >
    hello
  </div>
  <ul
    className="feature__list"
  >
    <li
      className="feature__item"
    >
      <div
        onClick={1}
      >
        (
        $NaN
        )
      </div>
    </li>
    <li
      className="feature__item"
    >
      <div
        onClick={1}
      >
        (
        $NaN
        )
      </div>
    </li>
  </ul>
</div>
`;
